### PR TITLE
WCAG upgrage - add an test & example file for alerts

### PR DESCRIPTION
**Pull request by shiva-sc**

### CODE REVIEW

**Review by duboisp — CHANGES_REQUESTED** [review]

File: common/alert/alerts-en.html [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+		{ "title": "GCWeb home", "link": "https://wet-boew.github.io/GCWeb/index-en.html" }
+	],
+	"dateModified": "2023-02-28"
+}
+---
+
+<h2>Alert with headings</h2>
+<section class="alert alert-danger">
+	<h3>Danger alert</h3>
+	<p>Optional alert details.</p>
```
Comment (duboisp) [inline common/alert/alerts-en.html:16]: This change to be applied to the other alert variant.

```suggestion
	<p>Alert details.</p>
```

Semantically we can not have a section that consist into being a single heading. An heading is always followed with content or inner sections.

**Review by duboisp — CHANGES_REQUESTED** [review]
I reviewed and after the small content change, we will be able to merge this change request.

* [ ] Remove the word "optional" for alert description

File: common/alert/alerts-en.html [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+</div>
+
+<h3>Code</h3>
+<pre><code>
+	//Sucess Alert with a heading
+	&lt;section <strong>class=&quot;alert alert-success&quot;</strong>&gt;
+	<strong>  &lt;h*&gt;...&lt;/h*&gt;</strong>
+	...
+	&lt;/section&gt;
+
```
Comment (duboisp) [inline common/alert/alerts-en.html:116]: Remove this code sample as it don't match the immediate preceding working example "Alerts without Headings"

File: common/alert/alerts-fr.html [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+	"title": "GCWeb Alert example",
+	"language": "fr",
+	"altLangPage": "alerts-en.html",
+	"breadcrumbs": [
+		{ "title": "GCWeb home", "link": "https://wet-boew.github.io/GCWeb/index-fr.html" }
+	],
+	"dateModified": "2023-02-28"
+}
+---
+
```
Comment (duboisp) [inline common/alert/alerts-fr.html:12]: To ensure we are conforming to "Language in page"
```suggestion
<p>Cette page nécessite une traduction.</p>
<div lang="en">
```

File: common/alert/alerts-fr.html [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+	</strong>&lt;section <strong>class=&quot;alert alert-danger&quot;</strong>&gt;
+	<strong>  &lt;h*&gt;...&lt;/h*&gt;</strong>
+	...
+	&lt;/section&gt;
+
+	//Danger Alert without a heading
+	&lt;div <strong>class=&quot;alert alert-danger&quot;</strong>&gt;
+		&lt;p&gt; Alert details. &lt;/p&gt;
+	&lt;/div&gt;
+</code></pre>
```
Comment (duboisp) [inline common/alert/alerts-fr.html:154]:
```suggestion
</code></pre>
</div>
```

**Review by duboisp — CHANGES_REQUESTED** [review]

File: common/alert/alerts-en.html [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+...
+&lt;/section&gt;
+
+//Info Alert without a heading
+&lt;div <strong>class=&quot;alert alert-info&quot;</strong>&gt;
+&lt;p&gt; Alert details. &lt;/p&gt;
+&lt;/div&gt;
+
+//Warning Alert with a heading<strong>
+</strong>&lt;section <strong>class=&quot;alert alert-warning&quot;</strong>&gt;
```
Comment (duboisp) [inline common/alert/alerts-en.html:125]: Remove the empty `strong`. This is a SC 1.3.1 issue.

```suggestion
//Warning Alert with a heading
&lt;section <strong>class=&quot;alert alert-warning&quot;</strong>&gt;
```

File: common/alert/alerts-fr.html [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+	</div>
+
+	<h3>Code</h3>
+	<pre><code>//Sucess Alert without a heading
+	&lt;div <strong>class=&quot;alert alert-success&quot;</strong>&gt;
+	&lt;p&gt; Alert details. &lt;/p&gt;
+	&lt;/div&gt;
+
+	//Info Alert with a heading<strong>
+	</strong>&lt;section <strong>class=&quot;alert alert-info&quot;</strong>&gt;
```
Comment (duboisp) [inline common/alert/alerts-fr.html:116]:
```suggestion
	//Info Alert with a heading
	&lt;section <strong>class=&quot;alert alert-info&quot;</strong>&gt;
```

File: common/alert/index.json-ld [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+	"title": {
+		"en": "Alert",
+		"fr": "Alertes"
+	},
+	"description": {
+		"en": "Alerts with custom styles.",
+		"fr": "Alertes avec des styles personalisés."
+	},
+	"modified": "2023-03-03",
+	"componentName": "tables",
```
Comment (duboisp) [inline common/alert/index.json-ld:18]: This component name is "alert". 
```suggestion
	"componentName": "alert",
```

Note: Having the wrong component name create a broken link in our main index page when it get generated by jekyll.

File: common/alert/alerts-en.html [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+	<h3>(Warning Title)</h3>
+	<p>(Warning content goes here <a href="#" class="alert-link">link text</a>.)</p>
+</section>
+<section class="alert alert-danger">
+	<h3>(Danger Title)</h3>
+	<p>(Danger content goes here <a href="#" class="alert-link">link text</a>.)</p>
+</section>
+<h3>Code</h3>
+<pre><code>&lt;section <strong>class=&quot;alert alert-success&quot;</strong>&gt;
+&lt;p&gt; Success content goes here &lt;a href=&quot;#&quot; <strong>class=&quot;alert-link&quot;</strong>&gt;link text&lt;/a&gt;.&lt;/p&gt;
```
Comment (duboisp) [inline common/alert/alerts-en.html:71]: The heading is missing here:

```suggestion
<pre><code>&lt;section <strong>class=&quot;alert alert-success&quot;</strong>&gt;
&lt;h3>(Success Title)&lt;/h3>
&lt;p&gt; Success content goes here &lt;a href=&quot;#&quot; <strong>class=&quot;alert-link&quot;</strong>&gt;link text&lt;/a&gt;.&lt;/p&gt;
```

File: common/alert/alerts-en.html [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+&lt;/section&gt;
+
+&lt;section <strong>class=&quot;alert alert-info&quot;</strong>&gt;
+&lt;p&gt; Info content goes here &lt;a href=&quot;#&quot; <strong>class=&quot;alert-link&quot;</strong>&gt;link text&lt;/a&gt;.&lt;/p&gt;
```
Comment (duboisp) [inline common/alert/alerts-en.html:75]:
```suggestion
&lt;h3>(Info Title)&lt;/h3>
&lt;p&gt; Info content goes here &lt;a href=&quot;#&quot; <strong>class=&quot;alert-link&quot;</strong>&gt;link text&lt;/a&gt;.&lt;/p&gt;
```

File: common/alert/alerts-en.html [hunk 10]
```diff
@@ -0,0 +1,144 @@
...
+&lt;/section&gt;
+
+&lt;section <strong>class=&quot;alert alert-warning&quot;</strong>&gt;
+&lt;p&gt; Warning content goes here &lt;a href=&quot;#&quot; <strong>class=&quot;alert-link&quot;</strong>&gt;link text&lt;/a&gt;.&lt;/p&gt;
```
Comment (duboisp) [inline common/alert/alerts-en.html:79]:
```suggestion
&lt;h3>(Warning Title)&lt;/h3>
&lt;p&gt; Warning content goes here &lt;a href=&quot;#&quot; <strong>class=&quot;alert-link&quot;</strong>&gt;link text&lt;/a&gt;.&lt;/p&gt;
```

File: common/alert/alerts-en.html [hunk 11]
```diff
@@ -0,0 +1,144 @@
...
+&lt;/section&gt;
+
+&lt;section <strong>class=&quot;alert alert-danger&quot;</strong>&gt;
+&lt;p&gt; Danger content goes here &lt;a href=&quot;#&quot; <strong>class=&quot;alert-link&quot;</strong>&gt;link text&lt;/a&gt;.&lt;/p&gt;
```
Comment (duboisp) [inline common/alert/alerts-en.html:83]:
```suggestion
&lt;h3>(Danger Title)&lt;/h3>
&lt;p&gt; Danger content goes here &lt;a href=&quot;#&quot; <strong>class=&quot;alert-link&quot;</strong>&gt;link text&lt;/a&gt;.&lt;/p&gt;
```

File: common/alert/alerts-en.html (hunk 5 above)
Comment (duboisp) [inline common/alert/alerts-en.html:125]: This code sample don't represent the HTML that is preceding as we should expect.
```suggestion
<h3>Code</h3>
<pre><code>&lt;h3> Success alert &lt;/h3>
&lt;div class="alert alert-success">
	&lt;p>(Success content here.)&lt;/p>
&lt;/div>

&lt;h3> Info alert &lt;/h3>
&lt;div class="alert alert-info">
	&lt;p> Info content here. &lt;/p>
&lt;/div>

&lt;h3> Warning alert &lt;/h3>
&lt;div class="alert alert-warning">
	&lt;p>(Warning content here.)&lt;/p>
&lt;/div>

&lt;h3> Danger alert &lt;/h3>
&lt;div class="alert alert-danger">
	&lt;p>(Danger content here.)&lt;/p>
&lt;/div>
```

**Review by duboisp — CHANGES_REQUESTED** [review]
There is a few change. Don't forget to apply the same similar change at the French version too.

**Comment by shiva-sc** [comment]
@duboisp Can you please let me know, if these are final changes to be made;

**Comment by duboisp** [comment]
@shiva-sc that is most likely the final changes to ensure the working example is conform to our web standard and illustrating the use of those style. There is always a risk, which are lowered at every review, to have additional change request. I don't see those as new requirement because published content must meet all our web standard at the time of publishing.

**Review by duboisp — APPROVED** [review]
Reviewed and tested locally,

It's coded and work as expected.

File: common/alert/alerts-en.html (hunk 5 above)
Comment (duboisp) [reply common/alert/alerts-en.html]: The requested change was applied

File: common/alert/alerts-en.html (hunk 11 above)
Comment (duboisp) [reply common/alert/alerts-en.html:87]: The requested change was applied

File: common/alert/alerts-en.html (hunk 10 above)
Comment (duboisp) [reply common/alert/alerts-en.html:82]: The requested change was applied

File: common/alert/alerts-en.html (hunk 9 above)
Comment (duboisp) [reply common/alert/alerts-en.html:77]: The requested change was applied

File: common/alert/alerts-en.html (hunk 8 above)
Comment (duboisp) [reply common/alert/alerts-en.html]: The requested change was applied

File: common/alert/alerts-fr.html (hunk 6 above)
Comment (duboisp) [reply common/alert/alerts-fr.html]: The requested change was applied

File: common/alert/alerts-en.html (hunk 5 above)
Comment (duboisp) [reply common/alert/alerts-en.html]: The requested change was applied

File: common/alert/alerts-en.html (hunk 2 above)
Comment (duboisp) [reply common/alert/alerts-en.html]: The requested change was applied

**Pull request merged**